### PR TITLE
TST: Add test to verify DataFrame's constructor doesn't convert Nones to strings on string columns

### LINKED ISSUE #32218: Inconsistent behavior when DataFrame with strings and None is created from lists or dictionary.
#### Code Sample

```python
import pandas as pd


pd.__version__

# df created from list, None is casted to str
df = pd.DataFrame(["1", "2", None], columns=["a"], dtype="str")
type(df.loc[2].values[0])


# Equivalent df created from dict, None remains NoneType
df = pd.DataFrame({"a": ["1", "2", None]}, dtype="str")
type(df.loc[2].values[0]))

```
#### Problem description

None is not casted consistently for DataFrames with None values and dtype set to str.

If DataFrame is created from list, then None is casted to str -> None -> "None".
If DataFrame is created from dict, then None remains NoneType -> None -> None.

IMO, the latter is the preferred behavior. I hope you consider this when continuing your work on string types and na types in future versions.  

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
machine          : AMD64
processor        : Intel64 Family 6 Model 142 Stepping 12, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.0.1
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 41.2.0
Cython           : None
pytest           : 5.3.5
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.1
IPython          : 7.12.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.16.0
pytables         : None
pytest           : 5.3.5
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by cgarciae**
- [x] closes #32218
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them

### whatsnew
Adds a very simple test to verify that `None`s are not casted to strings as `"None"` inside the DataFrame's constructor for string columns.

Questions:
* Should the test be more general?
* What is the proper location for the test?

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -2396,6 +2396,15 @@ def check_views():
             # assert b[0] == 0
             assert df.iloc[0, 2] == 0
 
+    def test_consistency_of_string_columns_with_none(self):
+        # df created from list, None is casted to str
+        df = DataFrame(["1", "2", None], columns=["a"], dtype="str")
+        type(df.loc[2].values[0])
```
Comment (jreback) [inline pandas/tests/frame/test_constructors.py:2402]: so this is not actually testing anything, instead you need to actually construct the expected frame and use assert_frame_equal. more to the point, I am pretty sure we have tests very similar to this. happy to take if we don't, but pls have a look (would take this if its slightly different than what we have, so pls give a look). we have a ton of tests so have to look around a bit.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_constructors.py (hunk 1 above)
Comment (cgarciae) [reply pandas/tests/frame/test_constructors.py]: Hey @jreback, by accident I didn't push the next commit with some actual asserts. 
However, if there is already a similar test that resolves #32218 we can reference it and close this PR.

Comment (jreback) [reply pandas/tests/frame/test_constructors.py]: yeah if you'd have a look around. I recall a similar issue to this but nothing referenced, pls have a look see at the existing tests (I know there are a lot)

Comment (jorisvandenbossche) [reply pandas/tests/frame/test_constructors.py]: It was fixed in 1.2.0 (but broken in 1.1.5), but I don't directly find anything in the 1.2.0 notes about this (and also didn't directly find an issue about it). So I think it is fine to add the test to be safe.

File: pandas/tests/frame/test_constructors.py [hunk 2]
```diff
@@ -2396,6 +2396,19 @@ def check_views():
             # assert b[0] == 0
             assert df.iloc[0, 2] == 0
 
+    def test_consistency_of_string_columns_with_none(self):
+        # df created from list, None is casted to str
+        df = DataFrame(["1", "2", None], columns=["a"], dtype="str")
+
+        assert isinstance(df.loc[0].values[0], str)
```
Comment (jorisvandenbossche) [inline pandas/tests/frame/test_constructors.py:2403]:
```suggestion
        assert isinstance(df.loc[0, "a"], str)
```

Comment (cgarciae) [reply pandas/tests/frame/test_constructors.py:2403]: Yeah, seems better, I think the issue was like that so it got carried.

**Comment by jreback** [comment]
closing as stale

**Pull request closed without merging**